### PR TITLE
Add parentheses in macros where needed for safety

### PR DESCRIPTION
**Pull request by nsajko**

### CODE REVIEW

**Comment by nsajko** [comment]
The problem is that parentheses were added even around macro operands which are types ...

**Comment by nsajko** [comment]
It seems that the test failure is not related to my commit, and it only happened in one configuration out of four on Travis. @radare @ret2libc @Maijin

**Comment by nsajko** [comment]
BTW, the main purpose of this change is to protect the intended operator precedence between macro operands and the surroundings of the place it gets substituted to, and getting better errors in case someone gives a non-expression operand to a macro that "expects" an expression.

File: binr/r2agent/r2agent.c [hunk 1]
```diff
@@ -102,7 +102,8 @@ int main(int argc, char **argv) {
 		const char *result = page_index;
 
 		rs = r_socket_http_accept (s, 0, timeout);
-		if (!rs) continue;
+		if (!rs) { continue;
```
Comment (ret2libc) [inline binr/r2agent/r2agent.c]: this is wrongly indented.

File: binr/rabin2/rabin2.c [hunk 2]
```diff
@@ -874,7 +874,8 @@ int main(int argc, char **argv) {
 			*p2++ = 0;
 			data = malloc (strlen (p2)+1);
 			datalen = r_hex_str2bin (p2, data);
-			if (datalen < 0) datalen = -datalen;
+			if (datalen < 0) { datalen = -datalen;
```
Comment (ret2libc) [inline binr/rabin2/rabin2.c]: also here

File: binr/r2agent/r2agent.c (hunk 1 above)
Comment (nsajko) [reply binr/r2agent/r2agent.c]: will fix

**Comment by nsajko** [comment]
I removed the other commits, we will have to do those after we are able to clang-format the code. Is the first (only remaining commit) OK?

**Comment by MaskRay** [comment]
If `shlr/*` are third-party code, they should be left untouched.

File: shlr/grub/fs/ufs.c [hunk 3]
```diff
@@ -48,13 +48,13 @@
 #define GRUB_UFS_VOLNAME_LEN	32
 
 /* Calculate in which group the inode can be found.  */
-#define UFS_BLKSZ(data) (grub_num_to_cpu32 (data->sblock.bsize, data->be))
+#define UFS_BLKSZ(data) (grub_num_to_cpu32 ((data)->sblock.bsize, (data)->be))
```
Comment (MaskRay) [inline shlr/grub/fs/ufs.c]: These third-party changes will make upgrading difficult.

Comment (nsajko) [reply shlr/grub/fs/ufs.c]: Fixed.

**Comment by nsajko** [comment]
I changed the commit message.

**Comment by radare** [comment]
You can just push -f

**Comment by nsajko** [comment]
>> I changed the commit message.
> You can just push -f

You mean git push --force? That is what I did...

**Comment by nsajko** [comment]
Any issues?

**Pull request merged**